src/openms/source/FORMAT/FASTAFile.cpp

```diff
@@ -39,184 +39,236 @@
+        // infile_ and outfile_ will close automatically when going out of scope. No need to do it explicitly here.
     }
 
-    if (infile_.is_open()) infile_.close(); // precaution
+    bool FASTAFile::readEntry_(std::string & id, std::string & seq)
+    {
+        std::streambuf* sb = infile_.rdbuf();
+        bool condition = true;
 
-    infile_.open(filename.c_str(), std::ios::binary | std::ios::in);
+        while(sb->sgetc() == '#')// Skip the header of PEFF files (http://www.psidev.info/peff)
+        {
+            infile_.ignore(numeric_limits<streamsize>::max(),'\n');
+        }
+        if(sb->sbumpc() == '>')//not saving '>'
+        {
+            while(condition)// reading the ID
+            {
+                int c = sb->sbumpc();// get and advance to next char
+                switch (c)
+                {
+                    case '\n'://ID finished
+                        condition = false;
+                        break;
+                    case std::streambuf::traits_type::eof():
+                        infile_.setstate(std::ios::eofbit);
+                        if (id.empty())
+                        { // only if we just started a new line, we set the is.fail() == true, ie. is == false
+                            infile_.setstate(std::ios::badbit);
+                        }
+                        return true;
+                    default:
+                        id += (char)c;
+                }
+            }
+        }
+        else return false;//was in wrong position for reading ID
+        if(id.empty()==true) return false;
 
-    // Skip the header of PEFF files (http://www.psidev.info/peff)
-    std::string line;
-    std::streampos firstline = 0;
-    while (TextFile::getLine(infile_, line))
-    {
-      if (!line.empty() && line[0] != '#')
-      {
-        break;
-      }
-      firstline = infile_.tellg();
+        condition = true;
+        while(condition)//reading the sequence
+        {
+            int c = sb->sbumpc();// get and advance to next char
+            switch (c)
+            {
+                case '\n':
+                    if (sb->sgetc() == '>') //reaching the beginning of the next protein-entry
+                    {
+                        condition = false;
+                    }
+                    break;
+                case '\r':
+                    if (sb->sgetc() == '\n') // peek current char
+                    {
+                        sb->sbumpc(); // consume it
+                    }
+                    break;
+                case ' ': //not saving white spaces
+                    break;
+                case '\t':
+                    break;
+                case std::streambuf::traits_type::eof():
+                    infile_.setstate(std::ios::eofbit);
+                    if (seq.empty())
+                    { // only if we just started a new line, we set the is.fail() == true, ie. is == false
```

Comment:
this code does not really make sense here (it did in the context of TextFile::getline()), because we never return `infile_` to the user, right?